File: android/app/src/main/java/com/studyapp/domain/model/Exam.kt
package com.studyapp.domain.model

import java.time.LocalDate
import java.time.temporal.ChronoUnit
import java.util.UUID

data class Exam(
    val id: Long = 0,
    val syncId: String = UUID.randomUUID().toString().lowercase(),
    val name: String,
    val date: Long,
    val note: String? = null,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis(),
    val deletedAt: Long? = null,
    val lastSyncedAt: Long? = null
) {
    val dateValue: LocalDate
        get() = LocalDate.ofEpochDay(date)

    fun daysRemaining(from: LocalDate = LocalDate.now()): Int {
        val start = from
        val end = dateValue
        return ChronoUnit.DAYS.between(start, end).toInt()
    }

    fun isPast(from: LocalDate = LocalDate.now()): Boolean =
        daysRemaining(from) < 0

    fun isToday(from: LocalDate = LocalDate.now()): Boolean =
        dateValue == from
}

File: ios/StudyApp/Domain/Models/Exam.swift
import Foundation

struct Exam: Identifiable, Codable, Hashable {
    var id: Int64 = 0
    var syncId: String = UUID().uuidString.lowercased()
    var name: String
    var date: Int64
    var note: String?
    var createdAt: Int64 = Date().epochMilliseconds
    var updatedAt: Int64 = Date().epochMilliseconds
    var deletedAt: Int64?
    var lastSyncedAt: Int64?

    var dateValue: Date {
        Date(epochDay: date)
    }

    func daysRemaining(from referenceDate: Date = Date()) -> Int {
        let calendar = Calendar.current
        let start = calendar.startOfDay(for: referenceDate)
        let end = calendar.startOfDay(for: dateValue)
        return calendar.dateComponents([.day], from: start, to: end).day ?? 0
    }

    func isPast(from referenceDate: Date = Date()) -> Bool {
        daysRemaining(from: referenceDate) < 0
    }
}
